app/controllers/segment_constraints/document_type_constraint.rb

```diff
@@ -1,6 +1,24 @@
+require "open-uri"
+
 class DocumentTypeConstraint
+  DOCUMENT_TYPE_YAML = "https://raw.githubusercontent.com/alphagov/rummager/master/config/govuk_index/migrated_formats.yaml".freeze
+
+  # Rummager's migrated formats yml doesn't contain all formats, so we need to hardcode these into our prototype
+  HARDCODED_FORMATS = ["residential_property_tribunal_decision", "task_list", "calculator", "field_of_operation", "html_publication", "imported", "ministerial_role", "organisation", "person", "policy_area", "services_and_information", "take_part", "topical_event", "topical_event_about_page", "working_group", "world_location", "worldwide_organisation", "corporate_information_page", "closed_consultation", "consultation_outcome", "esi_fund", "external_content", "national", "news_article", "official", "open_consultation", "statistics_announcement", "topic", "world_location_news_article"]
+
   def initialize
-    @doc_types = YAML.load_file(Rails.root.join("lib", "data", "formats.yml"))
+    @doc_types = load_document_types
+  end
+
+  def load_document_types
+    document_type_yaml_file = open(DOCUMENT_TYPE_YAML) {|f| f.read}
+    document_type_yaml_data = YAML::load(document_type_yaml_file)
+
+    document_type_yaml_data.each do |title, doc_formats|
```

Comment:
Like Python, Ruby lets you use `_` for unused variables. So you could write this line as:
`document_type_yaml_data.each do |_, doc_formats|`